src/main/java/hachi/Task.java

```diff
@@ -0,0 +1,63 @@
+package hachi;
+
+import java.time.LocalDate;
+
+/**
+ * Represents the individual tasks in the task list.
+ * Includes their completion status, name, and various methods.
+ */
+public class Task {
+    private boolean completed = false;
+    private String taskName = "";
+
+    public Task(String taskName) {
+        this.taskName = taskName;
+    }
+
+    /**
+     * Marks the task as completed.
+     */
+    public void mark() {
+        this.completed = true;
+    }
+
+    /**
+     * Marks the task as not completed.
+     */
+    public void unmark() {
+        this.completed = false;
+    }
+
+    /**
+     * Method to be overriden by tasks that have dates. For tasks without, always returns false.
+     *
+     * @return false
+     */
+    public boolean isDateWithinRange(LocalDate date) {
+        return false;
+    }
+
+    /**
+     * Returns true if string is within the task name.
+     *
```

Comment:
There are inconsistencies in your javaDoc, some have a line between description and params, some do not.